src/MQTT433gateway.cpp

```diff
@@ -27,251 +27,206 @@
-    if (otaAuth.verify(payload)) {
-      beatLED.on();
-      Debug(F("Start OTA update from: "));
-      DebugLn(otaURL);
-      rf.disableReceiver();
-      t_httpUpdate_return ret = ESPhttpUpdate.update(otaURL);
-      switch (ret) {
-        case HTTP_UPDATE_FAILED:
-          Debug(F("HTTP_UPDATE_FAILD Error ("));
-          Debug(ESPhttpUpdate.getLastError());
-          Debug(F("): "));
-          DebugLn(ESPhttpUpdate.getLastErrorString());
-          break;
-        case HTTP_UPDATE_NO_UPDATES:
-          DebugLn(F("HTTP_UPDATE_NO_UPDATES"));
-          break;
-        case HTTP_UPDATE_OK:
-          DebugLn(F("HTTP_UPDATE_OK"));  // may not called ESPhttpUpdate
-                                         // reboot the ESP?
-          ESP.restart();
-          break;
-      }
-      rf.enableReceiver();
-    } else {
-      DebugLn(F("OTA authentication failed!"));
+  rf = new RfHandler(settings, [](const String &protocol, const String &data) {
+    if (mqttClient) {
+      mqttClient->publishCode(protocol, data);
     }
-  }
+  });
+  rf->begin();
 }
 
-void rfCallback(const String &protocol, const String &message, int status,
-                int repeats, const String &deviceID) {
-  Debug(F("RF signal arrived ["));
-  Debug(protocol);
-  Debug(F("]/["));
-  Debug(deviceID);
-  Debug(F("] ("));
-  Debug(status);
-  Debug(F(") "));
-  Debug(message);
-  Debug(F(" ("));
-  Debug(repeats);
-  DebugLn(F(")"));
-
-  if (status == VALID) {
-    String topic = globalTopic + String(F("/recv/")) + protocol;
-    if (deviceID != nullptr) {
-      topic += String('/') + deviceID;
+void setupWebLog() {
+  if (webServer) {
+    if (settings.webLogLevel.length() > 0) {
+      Logger.addHandler(Logger.stringToLevel(settings.webLogLevel),
+                        webServer->logTarget());
+    } else {
+      Logger.removeHandler(webServer->logTarget());
     }
-    mqtt.publish(topic.c_str(), message.c_str(), true);
-  }
-  if (logMode) {
-    String topic = mainTopic + String(F("/log/")) + String(status) +
-                   String('/') + protocol;
-    mqtt.publish(topic.c_str(), message.c_str());
   }
 }
 
-void rfRawCallback(const uint16_t *pulses, int length) {
-  if (rawMode) {
-    String data = rf.pulseTrainToString(pulses, length);
-    if (data.length() > 0) {
-      Debug(F("RAW RF signal ("));
-      Debug(length);
-      Debug(F("): "));
-      Debug(data);
-      DebugLn();
-
-      mqtt.publish((mainTopic + F("/recvRaw")).c_str(), data.c_str());
-    }
+void setupWebServer(const Settings &s) {
+  webServer = new ConfigWebServer();
+
+  webServer->registerSystemCommandHandler(F("restart"), []() {
+    delay(100);
+    ESP.restart();
+  });
+  webServer->registerSystemCommandHandler(F("reset_wifi"), []() {
+    WiFi.disconnect(true);
+    delay(100);
+    ESP.restart();
+  });
+  webServer->registerSystemCommandHandler(F("reset_config"),
+                                          []() { settings.reset(); });
```

Comment:
Not working like this. Required a restart or more code in Settings.

Reply:
I think its helpful to have a new set of settings that can be changed before restarting. We can add a note that the new settings will get into full effect after restart.

Reply:
I did not get it. How should be the procedure? As user, if I click “Reset Config” I expect some change on the web interface.